### PR TITLE
Add the support to the scheduler for spreading pods of StatefulSets.

### LINKED ISSUE #41513: Add StatefulSet to set of controllers that SelectorSpreadPriority spreads
The SelectorSpreadPriority priority functions unfortunately has a hard-coded list of controller types whose pods it tries to spread across nodes. We never updated this list to include StatefulSet.

https://github.com/kubernetes/kubernetes/blob/3dbbd0bdf44cb07fdde85aa392adf99ea7e95939/plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go#L60
needs a "getPodStatefulSets()" function

Longer-term we really need a way to register scheduling-oriented controllers in a single place so we don't have to remember all the different places to update when we add a new controller.

(Theoretically we should also have Job in there...)

@kubernetes/sig-scheduling-feature-requests

### PR DESCRIPTION
**Pull request by bsalamat**
**What this PR does / why we need it**:
The scheduler SelectorSpread priority funtion didn't have the code to spread pods of StatefulSets. This PR adds StatefulSets to the list of controllers that SelectorSpread supports.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #41513

**Special notes for your reviewer**:

**Release note**:

```release-note
Add the support to the scheduler for spreading pods of StatefulSets.
```

### CODE REVIEW

**Comment by bsalamat** [comment]
I guess I signed the CLA! :-/

@davidopp

**Comment by davidopp** [comment]
@bsalamat you should follow the instructions at
https://github.com/kubernetes/kubernetes/wiki/CLA-FAQ

Ina nutshell (and these apply to anyone else who is an employee of a signed organization, just replace @google.com with your company's domain)
1. Add your @google.com email to github and make it "public"
2. Make sure your commits have the same @google.com address.
3. Sign in with github on the https://identity.linuxfoundation.org/projects/cncf webpage, after choosing to sign up as an employee.
4. Make sure you are using your @google.com email address on the create account page.

File: plugin/pkg/scheduler/algorithm/listers.go [hunk 1]
```diff
@@ -184,3 +185,42 @@ func (f FakeReplicaSetLister) GetPodReplicaSets(pod *v1.Pod) (rss []*extensions.
...
+		}
+		if selector.Matches(labels.Set(pod.Labels)) {
+			sss = append(sss, ss)
+		}
+	}
+	if len(sss) == 0 {
+		err = fmt.Errorf("Could not find StatefulSet for pod %s in namespace %s with labels: %v", pod.Name, pod.Namespace, pod.Labels)
+	}
+	return
+}
```
Comment (davidopp) [inline plugin/pkg/scheduler/algorithm/listers.go]: missing newline at end of file

**Review by davidopp — COMMENTED** [review]
Just one tiny nit

**Comment by bsalamat** [comment]
@davidopp 
I followed those instructions and the Linux foundation page shows me as authorized, but some of the commits from my laptops have been made with the default email address that git automatically assigns. Those are different from my Github email. Should I retract this PR and recommit the changes to a clean branch?

**Comment by davidopp** [comment]
Yeah, all of the commits have to be with the email address that signed the CLA.
Supposedly you can use `git commit --amend --author="Your Name <x@y.com>"` to change the email address on an existing commit but I've never tried it. If you can't get it to work then just create another PR.

File: plugin/pkg/scheduler/algorithm/listers.go (hunk 1 above)
Comment (bsalamat) [inline plugin/pkg/scheduler/algorithm/listers.go]: Done

**Pull request closed without merging**